### DISCUSSION TITLE
Support for Clang version 3.7.0

### DISCUSSION THREAD

**Discussion by bipin-rachaputi** [nlohmann/json#2896] [Q&A] [answered]
Hello,

 I do know that you support clang versions but when I try to add your library to my cmake using hunter (https://hunter.readthedocs.io/en/latest/packages/pkg/nlohmann_json.html) I get this weird generic error. It would be great if someone can help me narrowing down this error.

Thanks

```
Generate failed with: CMake Error in CMakeLists.txt:
  No known features for CXX compiler

  "Clang"

  version 3.7.0.


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Comment by nlohmann** [comment]
I did not provide the Hunter package myself. Maybe @jowr or @jothepro can help here.

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?

**Reply by bipin-rachaputi** [reply]
No thanks. I made some tweaks for work around, Thanks again

**Reply by nlohmann** [reply]
Thanks for the response! Is the fix something that it worth sharing?

**Comment by bipin-rachaputi** [comment, marked as answer]
Thanks for the reply. When I try to build with `ninja` it throws the following error and I guess that might be the blocker

`-- Build files have been written to: /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/Build/nlohmann_json-Debug-prefix/src/nlohmann_json-Debug-build
[13/16] Performing build step for 'nlohmann_json-Debug'

> ninja: no work to do.

[14/16] Performing install step for 'nlohmann_json-Debug'
[0/1] Install the project...
-- Install configuration: "Debug"
-- Up-to-date: /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/Install/include
-- Up-to-date: /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/Install/include/nlohmann
-- Up-to-date: /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/Install/include/nlohmann/json.hpp
-- Installing: /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/Install/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake
-- Installing: /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/Install/lib/cmake/nlohmann_json/nlohmann_jsonConfigVersion.cmake
-- Installing: /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/Install/lib/cmake/nlohmann_json/nlohmann_jsonTargets.cmake
-- Installing: /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/Install/lib/pkgconfig/nlohmann_json.pc
loading initial cache file /Users/bkumar/.hunter/_Base/60ffe7a/0985f04/6737201/Build/nlohmann_json/args.cmake
[16/16] Completed 'nlohmann_json-Debug'`
